pkg/lwip/contrib/sock/lwip_sock.c

```diff
@@ -513,6 +514,9 @@ ssize_t lwip_sock_send(struct netconn **conn, const void *data, size_t len,
         netbuf_delete(buf);
         return -ENOMEM;
     }
+    if (conn == NULL) {
+        noconn = 1;
```

Comment:
You could just move this assignment after [line 524](https://github.com/RIOT-OS/RIOT/pull/12927/commits/e671b00fc5d0372610c32a75b5fe799462a80b37#diff-9b8ebdee008416b1acde134ecd33c6f7R524) and safe an extra check.

Reply:
Mhmmm, but that line does more than only checking of conn == NULL?
    `if (((conn == NULL) || (*conn == NULL)) && (remote != NULL)) `
Would you prefer to split this conditional up in multiple parts, and then place the noconn assignment inside the block?

Reply:
Let me have another holistic view on the code...

Reply:
I think the code needs a lot more clean-up... Let me open an alternative PR.

Reply:
See https://github.com/RIOT-OS/RIOT/pull/12932. This code contained a lot of unnecessary second-level indirections.